processing/src/main/java/org/apache/druid/segment/DimensionHandler.java

```diff
@@ -100,6 +102,25 @@ default MultiValueHandling getMultivalueHandling()
    */
   DimensionIndexer<EncodedType, EncodedKeyComponentType, ActualType> makeIndexer(boolean useMaxMemoryEstimates);
 
+  /**
+   * @deprecated use {@link #makeMerger(String, IndexSpec, SegmentWriteOutMedium, ColumnCapabilities, ProgressIndicator, File, Closer)}
```

Comment:
Javadoc please about why this still exists. I think it's to avoid breaking `DimensionHandler` impls that exist in extensions.

Reply:
yea, i'll add, though I do plan to remove this shortly after this PR is merged, just trying to not be disruptive in the event we do a 31.1 (which i'd like to do as mentioned in PR description)